pkg/apis/discovery/validation/validation.go

```diff
@@ -172,24 +174,26 @@ func validatePorts(endpointPorts []discovery.EndpointPort, fldPath *field.Path)
 		return allErrs
 	}
 
-	portNames := sets.String{}
+	// Even though a sets.Set would be more idiomatic, we use a []string here to avoid
+	// extra allocations (especially since there are presumably only a few ports anyway).
```

Comment:
Another nit from me:
Could the reason behind this be added? Specifically the "Given how often EndpointSlices are updated" part from https://github.com/kubernetes/kubernetes/pull/130844#discussion_r2000274108

When I read the comment I was trying to figure out why we were trying to avoid extra allocations, the context of frequently updated EndpointSlices may be useful to someone that doesn't have that context

Reply:
That's fair.

I'm not going as far as demanding zero-allocation logic, but looking to
avoid obviously wasteful patterns in obviously frequent paths :)